rust/kernel/allocator.rs

```diff
@@ -12,12 +12,14 @@ pub struct KernelAllocator;
 
 unsafe impl GlobalAlloc for KernelAllocator {
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        // Safety: Returns a raw Pointer. Will return NULL in case of error.
         // `krealloc()` is used instead of `kmalloc()` because the latter is
         // an inline function and cannot be bound to as a result.
```

Comment:
I don't think `malloc` has particular safety requirements ([docs](https://www.kernel.org/doc/html/latest/core-api/mm-api.html#c.kmalloc)), so this can simply be
```suggestion
        // `krealloc()` is used instead of `kmalloc()` because the latter is
        // an inline function and cannot be bound to as a result.
        // SAFETY: FFI call.
```

Reply:
I also mentioned the C function name, `krealloc' in the new commit only FFI call might be confusing in places where the rust function is named differently.